src/pgduckdb_types.cpp

```diff
@@ -702,6 +702,153 @@ ConvertDecimal(const NumericVar &numeric) {
 	return (NumericIsNegative(numeric) ? -base_res : base_res);
 }
 
+duckdb::Value
+ConvertPostgresToDuckValue(Datum value, Oid postgres_type) {
+	switch (postgres_type) {
+	case BOOLOID:
+		return duckdb::Value::BOOLEAN(DatumGetBool(value));
+	case INT2OID:
+		return duckdb::Value::SMALLINT(DatumGetInt16(value));
+	case INT4OID:
+		return duckdb::Value::INTEGER(DatumGetInt32(value));
+	case INT8OID:
+		return duckdb::Value::BIGINT(DatumGetInt64(value));
+	case BPCHAROID:
+	case TEXTOID:
+	// TODO: Is this the correct place for JSONOID? I'm unable to use it in a
+	// test so I don't know.
+	// case JSONOID:
+	case VARCHAROID: {
+		return duckdb::Value((const char *)value);
+	}
+	case DATEOID:
+		return duckdb::Value::DATE(duckdb::date_t(static_cast<int32_t>(value + PGDUCKDB_DUCK_DATE_OFFSET)));
+	case TIMESTAMPOID:
+		return duckdb::Value::TIMESTAMP(
+		    duckdb::timestamp_t(static_cast<int64_t>(value + PGDUCKDB_DUCK_TIMESTAMP_OFFSET)));
+	case FLOAT4OID: {
+		return duckdb::Value::FLOAT(DatumGetFloat4(value));
+	}
+	case FLOAT8OID: {
+		return duckdb::Value::DOUBLE(DatumGetFloat8(value));
+	}
+	// case duckdb::LogicalTypeId::DECIMAL: {
```

Comment:
As you can see there's a few types that we support in other places that I didn't implement here yet. Also if you look at the added tests you'll see that there are a bunch of issues with floats and string arguments for some reason not working as expected in comparisons. 

In my opinion it's important to support them and fix those errors, but I don't think it necessarily needs to be part of this PR. This PR is already a bit larger than I'd like it to be. So I'd rather merge this soonish if we're happy with the general design, and then fix the remaining issues in follow up PRs. Especially since making DuckDB a singleton is a relatively large change in design that I feel would benefit from early use.

Reply:
This function is only used for binding? Maybe rename it, also we don't need to have commented out what is not supported (probably ticket should be created at this point).

Reply:
I renamed the function a bit and removed the commented out code. After we merge this I'll create a ticket to track support for the other types.